Tools/clinic/clinic.py

```diff
@@ -4710,9 +4710,14 @@ def bad_node(self, node):
 
         p = Parameter(parameter_name, kind, function=self.function, converter=converter, default=value, group=self.group)
 
-        if parameter_name in self.function.parameters:
+        names = [k.name for k in self.function.parameters.values()]
+        if parameter_name in names[1:]:
             fail("You can't have two parameters named " + repr(parameter_name) + "!")
-        self.function.parameters[parameter_name] = p
+        elif names and parameter_name == names[0] and c_name is None:
+            fail("Params named 'module' or 'self' need custom C names")
```

Comment:
Maybe this is better:

```suggestion
            fail(f"Parameter '{parameter_name}' requires a custom C name")
```

Reply:
Yes, it is better; I'll apply it.